### PR TITLE
curl new version 7.70.0 + fix build for Android on Windows

### PR DESCRIPTION
**Pull request by a4z**
Specify library name and version:  **libcurl/7.70.0**

This change also enables the build of curl for Android on Windows

The added os_build produces a warning, but seems to me required for being able to find out if we do Windows to Android compilation, 
if there is an more elegant way, please let me know
 

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libcurl/all/conanfile.py [hunk 1]
```diff
@@ -54,6 +54,11 @@ class LibcurlConan(ConanFile):
     def _is_mingw(self):
         return self.settings.os == "Windows" and self.settings.compiler != "Visual Studio"
 
+    @property
+    def _is_win_x_android(self):
+        return self.settings.os == "Android" and self.settings.os_build == "Windows"
```
Comment (uilianries) [inline recipes/libcurl/all/conanfile.py:59]: @jgsogo Can we detect it using the new cross-building feature?

Comment (a4z) [reply recipes/libcurl/all/conanfile.py:59]: just realize, maybe the `settings.os_build` is useless anyway for this purpose, 
`if platform.system() == "Windows":`  should tell me if I am on Windows
then I can remove the os_build from settings, and get rid of the warning ...

Comment (SSE4) [reply recipes/libcurl/all/conanfile.py:59]: so, Android build fails with configure on Windows? do you have an error message?

Comment (a4z) [reply recipes/libcurl/all/conanfile.py:62]: o dam, sorry , I just messed up this branch, too tired, 

and yes, on windows this configure script can not be executed 

I will close this PR and make a new one ....

**Pull request closed without merging**